src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,301 @@
+    PRINT_PARAM_STRING("preprocess_dataset") + " ."
+    "\n\n"
+    " Following arguments may be given " + PRINT_PARAM_STRING("lowercase") +
+    " to convert the dataset to lower case and " +
+    PRINT_PARAM_STRING("punctuation") + "for removing punctuation and "
+    + PRINT_PARAM_STRING("stopwords") + "for removing stopwords. Also the "
+    "dimension which contains the string dataset " +
+    PRINT_PARAM_STRING("dimension") + "."
+    "\n\n"
+    "So, a simple example where we want to preprocess string the dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN("column_delimiter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d", "\t");
+PARAM_STRING_IN("delimiter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("stopwordsfile", "File containing stopwords", "S", "");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+void createDataset(std::vector<std::vector<std::string>>& dataset,
+                   const std::string& filename,
+                   const std::string& column_delimiter)
+{
+  // Extracting the Contents of file
+  // File pointer
+  ifstream fin;
+  // Open an existing file
+  fin.open(filename);
+  if (!fin.is_open())
+  {
+    Log::Fatal << "Unable to open input file \n";
+  }
+  std::string line, word;
+  boost::string_view token, copy;
+  while (std::getline(fin, line))
+  {
+    stringstream streamLine(line);
+    dataset.emplace_back(std::vector<std::string>());
+    // delimeter[0] becase the standard function accepts char as input.
+    while (std::getline(streamLine, word, column_delimiter[0]))
+    {
+      dataset.back().push_back(word);
+    }
+  }
+}
+
+void checkForDigit(const std::string& column)
+{
+  for (auto& i : column)
+    if (!std::isdigit(i))
+    Log::Fatal << "Characters encountered, Please give only digits";
+}
+
+void getColumn(const std::vector<std::string>& temp_dimension,
+               std::set<size_t>& dimension)
```

Comment:
- We discussed that the output variable should be an unordered set since it supports constant time search while `std::set` provides logarithmic search.
- I think it is more natural to return the output rather than pass it as an argument i.e.
  `unordered_set<size_t> GetColumn(...)`
- The container contains a number of dimensions rather than a dimension. I suggest to use the name `dimensions`.